core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonMarshaller.java

```diff
@@ -27,11 +27,7 @@
 @SdkInternalApi
 public interface JsonMarshaller<T> extends Marshaller<T> {
```

Comment:
Nit: Can we make this a `@FunctionalInterface`?

Reply:
Done in 126fb115ec05f4648e133201d9132be970a2bd66. I also did the Query and Xml marshallers in 130d1c88f63d20779bcca1b9df25b7cf4a6382d2 because I like consistency.

Reply:
Nice!